chromatic_fitting.py

```diff
@@ -326,29 +469,47 @@ def setup_lightcurves(self):
         self.degree = self.parameters["p"].inputs['shape'] - 1
-        x = data.time.to_value()
 
-        with self.model:
-            if not hasattr(self, 'every_lightcurve'):
-                self.every_light_curve = {}
-            for i, w in enumerate(data.wavelength):
-                # compute polynomial of user-defined degree
-                poly = []
-                p = self.parameters["p"].get_prior(i)
-                for d in range(self.degree + 1):
-                    poly.append(p[d] * (x ** d))
-
-                # rd = RegexDict(dict(self.parameters))
-                # models_to_add = rd.get_matching('add_.*')
-                # for j, mod in enumerate(models_to_add):
-                # models_to_add = pm.math.sum([mod.get_prior() for mod in models_to_add],axis=-1)
-                if f"wavelength_{i}" not in self.every_light_curve.keys():
-                    self.every_light_curve[f"wavelength_{i}"] = pm.math.sum(poly, axis=0) #+ models_to_add #+ pm.math.sum(models_to_add, axis=0)
-                else:
-                    self.every_light_curve[f"wavelength_{i}"] += pm.math.sum(poly, axis=0)
+        if self.optimization == "separate":
+            models = self.pymc3_model
+            datas = [self.get_data(i) for i in range(self.data.nwave)]
+        else:
+            models = [self.pymc3_model]
+            datas = [self.get_data()]
+
+        if not hasattr(self, 'every_lightcurve'):
+            self.every_light_curve = {}
+
+        for j, (mod, data) in enumerate(zip(models, datas)):
+            with mod:
+                x = data.time.to_value()
+
+                for i, w in enumerate(data.wavelength):
+                    # compute polynomial of user-defined degree
+                    poly = []
+                    p = self.parameters["p"].get_prior(i+j)
+                    for d in range(self.degree + 1):
+                        poly.append(p[d] * (x ** d))
+
+                    # rd = RegexDict(dict(self.parameters))
+                    # models_to_add = rd.get_matching('add_.*')
+                    # for j, mod in enumerate(models_to_add):
+                    # models_to_add = pm.math.sum([mod.get_prior() for mod in models_to_add],axis=-1)
+                    if f"wavelength_{i+j}" not in self.every_light_curve.keys():
+                        self.every_light_curve[f"wavelength_{i+j}"] = pm.math.sum(poly,
+                                                                                axis=0)  # + models_to_add #+ pm.math.sum(models_to_add, axis=0)
+                    else:
+                        self.every_light_curve[f"wavelength_{i+j}"] += pm.math.sum(poly, axis=0)
+
+    def polynomial_model(self, poly_params):
+        data = self.get_data()
+        poly = []
+        for d in range(self.degree + 1):
+            poly.append(poly_params['p'][d] * (data.time.to_value() ** d))
+        return (np.sum(poly, axis=0))
+
```

Comment:
I know it's a perfect subset of `PolynomialModel`, but for convenience it might be nice to have a `ConstantModel` that's just a 0-order polynomial, so we can specify `Fixed`, `Fitted`, or `WavelengthFitted `values without having to worry about providing an accurate `shape` option.

Reply:
I think this could still work in PolynomialModel if I replaced in the code where I use `.inputs['shape']` with something more clever, so if the user inputs something without a shape (which would have to be 1D) then it would work without defining the shape?

Reply:
I think that might be nice, for it to be easy/obvious to use PolynomialModel as a simple constant.